### DISCUSSION TITLE
[SOLVED] cmake issue related to SQLite when building v4.5.1

### DISCUSSION THREAD

**Discussion by Bro-Account** [martinrotter/rssguard#1126] [General]
Hi, I have some issues again compiling the new version. I was successfully able to compile and run it on my

Ubuntu 23.04 (Raspberry) 
cmake 3.25.1
sqlite3 3.40.1

after installing libsqlite3-dev. But the same procedure doesn't work on my

Raspberry PI OS
Debian GNU/Linux 11 (bullseye)
cmake 3.18.4
sqlite3 3.34.1

On the reference installation, I get an additional "Found SQLite3" line at the output, that I don't get with Raspberry PI OS.

```
mypi@mypi-pibuntu:~/Documents/rssguard/build-dir$ cmake -DCMAKE_BUILD_TYPE="Release" ../rssguard-4.5.1
-- Found SQLite3: /usr/include (found version "3.40.1")
-- Using system SQLite3 3.40.1.
-- Feedly client ID/secret variables are not set. Disabling official support.
-- Gmail client ID/secret variables are not set. Disabling official support.
-- Inoreader client ID/secret variables are not set. Disabling official support.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/mypi/Documents/rssguard/build-dir
```

```
myotherpi@raspberrypi400:~/Documents/rssguard/build-dir $ cmake -DCMAKE_BUILD_TYPE="Release" ../rssguard-4.5.1
-- Using system SQLite3 3.34.1.
-- Feedly client ID/secret variables are not set. Disabling official support.
-- Gmail client ID/secret variables are not set. Disabling official support.
-- Inoreader client ID/secret variables are not set. Disabling official support.
CMake Error at src/librssguard/CMakeLists.txt:663 (target_include_directories):
  target_include_directories called with invalid arguments


-- Configuring incomplete, errors occurred!
See also "/home/myotherpi/Documents/rssguard/build-dir/CMakeFiles/CMakeOutput.log".

```

I checked for the /usr/include/sqlite3.h file, which is there.

I checked the CMakeLists code. I found this part.

```
# Add sqlite.
if(USE_SYSTEM_SQLITE)
  find_package(SQLite3)

  if(SQLite3_FOUND)
    # Include directory and library are linked to below.
    message(STATUS "Using system SQLite3 ${SQLite3_VERSION}.")
```

Even though there is no "Found" message, the result should still be positive, otherwise it would not go into the following if block.

The actual error happens at this point ("target_include_directories called with invalid arguments") and is probably related to ${SQLite3_INCLUDE_DIRS}.

```
if(SQLite3_FOUND)
  target_include_directories(rssguard AFTER
    PRIVATE
    ${SQLite3_INCLUDE_DIRS}
  )
```

Here I'm stuck with my troubleshooting. I tried to change the option in the CMakeLists.txt file to

`option(USE_SYSTEM_SQLITE "Use system-wide SQLite3 library." OFF)`

but weirdly, I still get the same result, including `Using system SQLite3 3.34.1.`

Any ideas what might be the issue?

Thanks!

**Comment by martinrotter** [comment]
https://github.com/martinrotter/rssguard/blob/f6b89bd1c242583605133dceccc69f19714f4bf7/src/librssguard/CMakeLists.txt#L522


Compile with that option explicitly set to OFF.

**Reply by Bro-Account** [reply]
Thanks, if(OFF) worked.
